worker/handler/handler.go

```diff
@@ -277,6 +360,20 @@ func (h *Handler) RunStart() (ch *sb.Channel, err error) {
 
 	// create sandbox if needed
 	if h.sandbox == nil {
+		// if estimated mem usage exceeds the hard limit, the handler would queue itself
+        hms.sbCreatingLock.Lock()
+        for hms.sbHardLimit != 0 && (getMemUsage() + (hms.numSbCreating + 1) * hms.avgHandlerMemUsage > hms.sbHardLimit) {
+            hms.sbCreatingLock.Unlock()
+			if h.sbCreatingChan == nil {
+				h.sbCreatingChan = make(chan bool, 1)
+			}
+			hms.sbCreatingQueue <- h
+			<-h.sbCreatingChan // waiting for wake-up signal
+            hms.sbCreatingLock.Lock()
```

Comment:
Weird indenting in a few places

Reply:
Sorry for the confusion about this, will fix in the next commit.